hamilton/function_modifiers/macros.py

```diff
@@ -913,24 +941,88 @@ def B(...):
     3. You want to use the same function multiple times, but with different parameters -- while `@does`/`@parameterize` can
     do this, this presents an easier way to do this, especially in a chain.
 
-    The rules for chaining nodes as the same as for pipe.
+    The rules for chaining nodes are the same as for pipe_input.
+
+    For extra control in case of multiple output nodes, for example after extract_field / extract_columns we can also specify the output node that we wish to mutate.
+    The following apply *A* to all fields while *B* only to "field_1"
+
+    .. code-block:: python
+        :name: Simple @pipe_output example targeting specific nodes
+
+        @extract_columns("col_1", "col_2")
+        def A(...):
+            return ...
+
+        def B(...):
+            return ...
+
+
+         @pipe_output(
+            step(A),
+            step(B).on_output("field_1"),
+        )
+        @extract_fields(
+                {"field_1":int, "field_2":int, "field_3":int}
+        )
+        def foo(a:int)->Dict[str,int]:
+            return {"field_1":1, "field_2":2, "field_3":3}
+
+    We can also do this on the global level (but cannot do on both levels at the same time). The following would apply function *A* and function *B* to only "field_1" and "field_2"
+
+    .. code-block:: python
+        :name: Simple @pipe_output targeting specific nodes local
+
+        @pipe_output(
+            step(A),
+            step(B),
+            on_output = ["field_1","field_2]
+        )
+        @extract_fields(
+                {"field_1":int, "field_2":int, "field_3":int}
+        )
+        def foo(a:int)->Dict[str,int]:
+            return {"field_1":1, "field_2":2, "field_3":3}
     """
 
+    @classmethod
+    def _validate_single_target_level(cls, target: base.TargetType, transforms: Tuple[Applicable]):
+        """We want to make sure that target gets applied on a single level.
+        Either choose for each step individually what it targets or set it on the global level where
+        all steps will target the same node(s).
+        """
+        if target is not None:
+            for transform in transforms:
+                if transform.target is not None:
+                    raise SingleTargetError("Cannot have target set on pipe_output and step level.")
+
     def __init__(
         self,
         *transforms: Applicable,
         namespace: NamespaceType = ...,
+        on_output: base.TargetType = None,
         collapse=False,
         _chain=False,
     ):
         """Instantiates a `@pipe_output` decorator.
 
+        Warning: if there is a global pipe_output target, the individual Applicable.target only chooses
+        from the subset pre-selected from the global pipe_output target. Leave global pipe_output target
+        empty if you want to choose between all the nodes on the individual Applicable level.
```

Comment:
This will break, right? In the `_validate_single_target_level` function? If so mention that here.